sdk/tests/conformance/extensions/webgl-compressed-texture-astc.html

```diff
@@ -1678,9 +1932,101 @@
+        // sampling result to magenta.
+
+        nanToMagentaProgram2D = wtu.setupSimpleTextureProgramESSL300(gl, undefined, undefined,
+           `#version 300 es
+            precision mediump float;
+            uniform sampler2D tex;
+            in vec2 texCoord;
+            out vec4 out_color;
+            void main() {
+                vec4 c = texture(tex, texCoord);
+                if (all(isnan(c))) {
+                    out_color = vec4(1, 0, 1, 1);
+                } else {
+                    out_color = c;
+                }
+            }`
+        );
+
+        nanToMagentaProgram2DArray = wtu.setupSimpleTextureProgramESSL300(gl, undefined, undefined,
+           `#version 300 es
+            precision mediump float;
+            uniform mediump sampler2DArray tex;
+            in vec2 texCoord;
+            out vec4 out_color;
+            void main() {
+                vec4 c = texture(tex, vec3(texCoord, 0));
+                if (all(isnan(c))) {
+                    out_color = vec4(1, 0, 1, 1);
+                } else {
+                    out_color = c;
+                }
+            }`
+        );
+
+        nanToMagentaProgram3D = wtu.setupSimpleTextureProgramESSL300(gl, undefined, undefined,
+           `#version 300 es
+            precision mediump float;
+            uniform mediump sampler3D tex;
+            in vec2 texCoord;
+            out vec4 out_color;
+            void main() {
+                vec4 c = texture(tex, vec3(texCoord, 0));
+                if (all(isnan(c))) {
+                    out_color = vec4(1, 0, 1, 1);
+                } else {
+                    out_color = c;
+                }
+            }`
+        );
+
+        program2DArray = wtu.setupSimpleTextureProgramESSL300(gl, undefined, undefined,
+           `#version 300 es
+            precision mediump float;
+            uniform mediump sampler2DArray tex;
+            in vec2 texCoord;
+            out vec4 out_color;
+            void main() {
+                out_color = texture(tex, vec3(texCoord, 0));
+            }`
+        );
+
+        program3D = wtu.setupSimpleTextureProgramESSL300(gl, undefined, undefined,
+           `#version 300 es
+            precision mediump float;
+            uniform mediump sampler3D tex;
+            in vec2 texCoord;
+            out vec4 out_color;
+            void main() {
+                out_color = texture(tex, vec3(texCoord, 0));
+            }`
+        );
+    }
+
+    // Query supported profiles and test ASTC texture with appropriate formats
+    var profiles = ext.getSupportedProfiles();
+    if (profiles.length > 0) {
+        // Check for HDR support first, since it affects sliced 3D support for LDR
+        if (profiles.indexOf("hdr") != -1) {
+            hasHdr = true;
+            testHDRTextures();
+        } else {
+            testPassed("HDR profile is not supported.");
+        }
+
+        if (profiles.indexOf("ldr") != -1) {
```

Comment:
Instead of checking for an empty array, and also requiring ldr, you can simply require ldr to be present, and that guarantees we don't have an empty array.